aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py

```diff
@@ -1,25 +1,77 @@
-from .utils import make_valid_data
+from .utils import (
+    make_valid_data,
+    create_arn as arn,
+    client_array_operation,
+    set_required_access_v2,
+    transformation,
+)
 from .registry import RegisteredResourceCollector
+from collections import namedtuple
+from schematics import Model
+from schematics.types import StringType
+from copy import deepcopy
+
+
+def create_arn(region=None, account_id=None, resource_id=None, **kwargs):
+    return arn(resource="route53", region="", account_id=account_id, resource_id="domain/" + resource_id)
+
+
+DomainData = namedtuple("DomainData", ["domain", "tags"])
+
+
+class Domain(Model):
+    DomainName = StringType(required=True)
 
 
 class Route53DomainCollector(RegisteredResourceCollector):
     API = "route53domains"
     API_TYPE = "global"
     COMPONENT_TYPE = "aws.route53.domain"
 
+    @set_required_access_v2("route53domains:ListTagsForDomain")
+    def collect_tags(self, domain_name):
+        return self.client.list_tags_for_domain(DomainName=domain_name).get("TagList", [])
+
+    @set_required_access_v2("route53domains:GetDomainDetail")
+    def collect_domain_description(self, domain_name):
+        return self.client.get_domain_detail(DomainName=domain_name) or {}
+
+    def collect_domain(self, domain_data):
+        domain_name = domain_data.get("DomainName", "")
+        data = deepcopy(domain_data)
```

Comment:
I think the `deepcopy` is unnecessary, we're not sharing this dict right?

Reply:
In general I've been taught that you should not mutate a reference that is passed into a function, but it won't break anything if I don't do it.

I don't mind either way, let me know and I'll remove it